### ISSUE TITLE
WGSL: Unrecognized single-token diagnostic rule names are an error, but it should be a warning

### ISSUE THREAD

**Issue by ErichDonGubler** [gfx-rs/wgpu#6460]
**Description**

Initial support for `diagnostic` filtering rules will land/has landed in #6456. With it, Naga's WGSL input will reject the following shader:

```wgsl
diagnostic(off, asdf);
```

…with this error:

```
error: invalid `diagnostic(…)` rule name
  ┌─ ../in.wgsl:1:17
  │
1 │ diagnostic(off, asdf);
  │                 ^^^^ not a valid diagnostic rule name
  │
  = note: See available trigger rules at <https://www.w3.org/TR/WGSL/#filterable-triggering-rules>.
```

However, the WGSL standard [states](https://www.w3.org/TR/WGSL/#filterable-triggering-rules):

> Using an unrecognized triggering rule consisting of a single [diagnostic name-token](https://www.w3.org/TR/WGSL/#diagnostic-name-token) should trigger a warning from the user agent.

**Repro steps**

See above.

**Expected vs observed behavior**

warning plz

**Extra materials**

\- 

**Platform**

\-

**Comment by ErichDonGubler** [comment]
Currently blocked on <https://github.com/gfx-rs/wgpu/issues/6458>.

**Comment by ErichDonGubler** [comment]
This was actually resolved in #6456 at review time ([convo link](https://github.com/gfx-rs/wgpu/pull/6456#discussion_r1816454048)). This is no longer valid.
